File: be/pwandora/src/main/java/com/ssafy/pwandora/acceptance/entity/AcceptanceChecklist.java
package com.ssafy.pwandora.acceptance.entity;

import com.ssafy.pwandora.pwa.entity.Pwa;
import jakarta.persistence.*;
import lombok.*;

/* PWA 검수 체크리스트 엔티티 */
@Entity
@Table(name = "acceptance_checklist")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AcceptanceChecklist {

    @Id
    @Column(name = "pwa_id")
    private Integer pwaId;

    @MapsId
    @OneToOne
    @JoinColumn(name = "pwa_id")
    private Pwa pwa;

    @Column(name = "crawled_status")
    private String crawledStatus;

    @Column(name = "display_status")
    private String displayStatus;

    @Column(name = "screenshot_status")
    private String screenshotStatus;

    @Column(name = "ai_suggestion_status")
    private String aiSuggestionStatus;

    @Column(name = "ai_censor_status")
    private String aiCensorStatus;

    @Column(name = "packaging_status")
    private String packagingStatus;

    @Column(name = "lighthouse_status")
    private String lighthouseStatus;

    @Column(name = "security_status")
    private String securityStatus;
}

File: crawl/pwandora/models/acceptance_checklist.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('acceptance_checklist', {
    pwa_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true,
      comment: "PWA 테이블의 ID를 참조",
      references: {
        model: 'pwa',
        key: 'id'
      }
    },
    crawled_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    display_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    screenshot_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    ai_suggestion_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    ai_censor_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    packaging_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    lighthouse_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    },
    security_status: {
      type: DataTypes.STRING(255),
      allowNull: true
    }
  }, {
    sequelize,
    tableName: 'acceptance_checklist',
    timestamps: false,
    indexes: [
      {
        name: "PRIMARY",
        unique: true,
        using: "BTREE",
        fields: [
          { name: "pwa_id" },
        ]
      },
    ]
  });
};
